### PR TITLE
Fix a wrong method usage

### PR DESCRIPTION
**Pull request by leofeyer**
Follow-up on #3659.

### CODE REVIEW

File: core-bundle/src/Resources/contao/modules/Module.php [hunk 1]
```diff
@@ -257,7 +257,7 @@ abstract protected function compile();
 	 */
 	protected function getResponseCacheTags(): array
 	{
-		return array(System::getContainer()->get(EntityCacheTags::class)->getTagForModelClass($this->objModel));
+		return array(System::getContainer()->get(EntityCacheTags::class)->getTagForModelInstance($this->objModel));
```
Comment (ausi) [inline core-bundle/src/Resources/contao/modules/Module.php:265]: Should we also check that `$this->objModel` is not null?

**Review by m-vo — APPROVED** [review]

File: core-bundle/src/Resources/contao/modules/Module.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/modules/Module.php:265]: We probably should. 👍

Comment (leofeyer) [reply core-bundle/src/Resources/contao/modules/Module.php:265]: Changed in b7ca2d1d0bb00eb434a5321153cdec05eec9f120.

Comment (m-vo) [reply core-bundle/src/Resources/contao/modules/Module.php:265]: How can this be null? The property is annotated with `@var ModuleModel` as well. On the other hand: the constructor type hints are wrong as well.

File: core-bundle/src/Resources/contao/modules/Module.php [hunk 2]
```diff
@@ -257,7 +257,12 @@ abstract protected function compile();
 	 */
 	protected function getResponseCacheTags(): array
 	{
-		return array(System::getContainer()->get(EntityCacheTags::class)->getTagForModelClass($this->objModel));
+		if (!$this->objModel instanceof Model)
```
Comment (m-vo) [inline core-bundle/src/Resources/contao/modules/Module.php]: Why don't we check for `null` here? 
We should probably also adjust the typehint in L104 and return type of `getModel` to `?ModuleModel`, no?

Comment (leofeyer) [reply core-bundle/src/Resources/contao/modules/Module.php]: Because the `getTagForModelInstance()` method expects a `Model` object.

Comment (m-vo) [reply core-bundle/src/Resources/contao/modules/Module.php]: Yes sure, but how can `$this->objModel` be anything else than a `Model` or `null` (if not defined). It gets set in the constructor where the type is checked.

Comment (leofeyer) [reply core-bundle/src/Resources/contao/modules/Module.php]: Changed in 60f279d3d835d53335127c8ac0e06be80ffc4d8a.

**Comment by m-vo** [comment]
I just realized, that `ModuleArticle` will probably have to get this change as well…

**Comment by leofeyer** [comment]
That's out of scope of this PR, though.

**Comment by m-vo** [comment]
> That's out of scope of this PR, though.

IMO it's related as it was also introduced in #3659. See https://github.com/leofeyer/contao/pull/1

**Review by ausi — APPROVED** [review]

**Comment by leofeyer** [comment]
True. Added in 1cbd271b0396e2c76197cf4a9d6ae5be58277cbe.

**Comment by leofeyer** [comment]
This actually applies to even more classes such as e.g. the `Hybrid` class. And there are quite a few classes that use `$this->objModel` without checking `null` such as e.g. the `ContentElement` class. Therefore, I guess we can safely assume that `Module::$objModel` is never supposed to be `null`.

So IMHO we should add an exception in line 139 instead:

```php
if (!$objModule instanceof Model)
{
    throw new \LogicException('No model given');
}
```

**Review by fritzmg — APPROVED** [review]

**Review by m-vo — APPROVED** [review]

**Comment by aschempp** [comment]
This change will prevent a content element etc. from being created from a database result, which was the behaviour we had before models. That's why we check for a model in the first if-statement. Is that intentional, maybe it should only be done in Contao 5?

**Comment by leofeyer** [comment]
You are absolutely right. Fixed in #3735.

**Pull request merged**